This dataset contains over 39,000 data points about weather conditions in London in 2015, including humidity, temperature, precipitation, and others. We can use the dataset to better understand the weather conditions year-round and plan for trips.  

In [1]:
import pandas as pd
import numpy as np

In [3]:
data=pd.read_csv("C:/Users/carol/Desktop/London_Weathers.csv")
print(data.head())

  timestamp  Hour  Month  Temperature  Humidity  Pressure  Precipitation  \
0  1/1/2015     0      1    -0.839471        81    1035.6            0.0   
1  1/2/2015     0      1     0.210529        70    1036.2            0.0   
2  1/3/2015     0      1     2.050528        94    1037.9            0.0   
3  1/4/2015     0      1     9.910529        88    1024.1            0.0   
4  1/5/2015     0      1     0.290529        86    1039.9            0.0   

   Snowfall  Basel Cloud Cover Total  Basel Cloud Cover High [high cld lay]  \
0       0.0                      0.0                                      0   
1       0.0                      0.0                                      0   
2       0.0                    100.0                                     10   
3       0.0                    100.0                                      0   
4       0.0                      0.0                                      0   

   Basel Cloud Cover Medium [mid cld lay]  \
0                      

In [6]:
temp=data['Temperature']
average_temp= np.mean(temp)
temperature_var= np.var(temp)
temperature_standard_deviation= np.std(temp)

print(average_temp)
print(temperature_var)
print(temperature_standard_deviation)

12.660235575676257
58.28864318785136
7.634699940917872


The average temperature in London is 12.7 degree celsius, with a 7.6 degree standard deviation.

In [8]:
june= data[data.Month == 6]["Temperature"]

july= data[data.Month == 7]["Temperature"]

june_mean= np.mean(june)
print(june_mean)

july_mean= np.mean(july)
print(july_mean)

june_std= np.std(june)
print(june_std)

july_std= np.std(july)
print(july_std)

19.300348608333334
23.308217354166665
4.628820527668157
5.6233837861476


Now a client is thinking about travelling to London in June or July. After analysis, it seems like it's warmer in July on average, but the variation in temperature is also larger. 

In [13]:
for i in range(1, 13):
  month = data[data.Month== i]["Temperature"]
  print("The mean temperature in month "+str(i) +" is "+ str(np.mean(month)))
  print("The standard deviation of temperature in month "+str(i) +" is "+ str(np.std(month))+"\n")

for i in range(1, 13):
  month = data[data.Month== i]["Humidity"]
  print("The mean humidity in month "+str(i) +" is "+ str(np.mean(month)))
  print("The standard deviation of humidity in month "+str(i) +" is "+ str(np.std(month))+"\n")

for i in range(1, 13):
  month = data[data.Month== i]["Precipitation"]
  print("The mean precipitation in month "+str(i) +" is "+ str(np.mean(month)))
  print("The standard deviation of precipitation in month "+str(i) +" is "+ str(np.std(month))+"\n")

for i in range(1, 13):
  month = data[data.Month== i]["Basel Sunshine Duration"]
  print("The mean sunshine duration in month "+str(i) +" is "+ str(np.mean(month)))
  print("The standard deviation of sunshine duration in month "+str(i) +" is "+ str(np.std(month))+"\n")

for i in range(1, 13):
  month = data[data.Month== i]["Basel Wind Speed [10 m]"]
  print("The mean wind speed in month "+str(i) +" is "+ str(np.mean(month)))
  print("The standard deviation of wind speed in month "+str(i) +" is "+ str(np.std(month))+"\n")



The mean temperature in month 1 is 4.299937261465054
The standard deviation of temperature in month 1 is 3.5835545217740883

The mean temperature in month 2 is 3.3784304155505955
The standard deviation of temperature in month 2 is 3.246146918285593

The mean temperature in month 3 is 8.225932053954299
The standard deviation of temperature in month 3 is 3.853972064858433

The mean temperature in month 4 is 11.398001162000002
The standard deviation of temperature in month 4 is 5.125686499182164

The mean temperature in month 5 is 15.261873152419355
The standard deviation of temperature in month 5 is 4.4310203781932955

The mean temperature in month 6 is 19.300348608333334
The standard deviation of temperature in month 6 is 4.628820527668157

The mean temperature in month 7 is 23.308217354166665
The standard deviation of temperature in month 7 is 5.6233837861476

The mean temperature in month 8 is 21.863593726478495
The standard deviation of temperature in month 8 is 5.311961621778022

Th

Now expanding to look at every month across various weather metrics such as temperature, humidity, wind speed, precipitation, and sunshine duration. From examining various metrics, July is the most ideal month for travelling in London, because it's the warmest and the least humid, with the lowest precipitation and the longest sunshine duration across summer months.